scipy/signal/_peak_finding.py

```diff
@@ -709,77 +709,6 @@ def _select_by_peak_threshold(x, peaks, tmin, tmax):
     return keep, stacked_thresholds[0], stacked_thresholds[1]
 
 
-# Code for _select_by_peak_distance was adapted from
-# https://github.com/demotu/BMC/blob/master/functions/detect_peaks.py
-# by Marcos Duarte under the MIT license:
-#
-#     Copyright (c) 2013 Marcos Duarte
-#
-#     Permission is hereby granted, free of charge, to any person
-#     obtaining a copy of this software and associated documentation
-#     files (the "Software"), to deal in the Software without
-#     restriction, including without limitation the rights to use,
-#     copy, modify, merge, publish, distribute, sublicense, and/or sell
-#     copies of the Software, and to permit persons to whom the
-#     Software is furnished to do so, subject to the following
-#     conditions:
-#
-#     The above copyright notice and this permission notice shall be
-#     included in all copies or substantial portions of the Software.
-#
-#     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-#     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
-#     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-#     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
-#     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
-#     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-#     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
-#     OTHER DEALINGS IN THE SOFTWARE.
```

Comment:
You should probably keep this notice with the Cython code. Even though the language has changed the algorithm came from there.

Reply:
I'm kind of surprised by that statement and I'd argue that, depending on where you draw the line, the two approaches have the same goal but use different ways (vectorized vs iterative) to reach it. 
If it matters: while writing it I didn't have the feeling of adapting the former but formulating a new more direct idea. I mean there are only a few algorithmic ways to implement this kind of simple thing.

Anyway if you still feel I should include the notice its fine by me. I just thought it would be justified to get rid of the text blob.